File: 206_ReverseLinkedList/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    if(head == NULL){
        return head;
    }
    struct ListNode* prev = head;
    struct ListNode *curr = head->next;
    if(curr == NULL){
        return head;
    }
    struct ListNode *save = curr->next;
    prev->next = NULL;
    while(curr != NULL){
        curr->next = prev;
        prev = curr;
        curr = save;
        if(save != NULL){
            save = save->next;
        }
    }
    return prev;   
}

File: 206_ReverseLinkedList/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

"""
Same code logic applies as the C solution, just implemented in Python syntax. (28/28 cases passed)
"""
class Solution(object):
    def reverseList(self, head):
        if (head == None):
            return head
        curr = head.next
        if (curr == None):
            return head
        save = curr.next
        head.next = None
        while(curr != None):
            curr.next = head
            head = curr
            curr = save
            if (save != None):
                save = save.next
        return head

        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        
